component/config_test.go

```diff
@@ -419,28 +417,3 @@ func TestNewType(t *testing.T) {
 		})
 	}
 }
-
-type configWithEmbeddedStruct struct {
-	String string `mapstructure:"string"`
-	Num    int    `mapstructure:"num"`
-	embeddedUnmarshallingConfig
-}
-
-type embeddedUnmarshallingConfig struct {
-}
-
-func (euc *embeddedUnmarshallingConfig) Unmarshal(_ *confmap.Conf) error {
-	return nil // do nothing.
-}
-func TestStructWithEmbeddedUnmarshaling(t *testing.T) {
```

Comment:
Why are we removing this test? Is it being move somewhere else?

Reply:
It's moved under confmap.

Reply:
brought it back with a few changes so it passes!